packages/util/src/bytes.ts

```diff
@@ -1,14 +1,54 @@
-import { assertIsArray, assertIsBuffer, assertIsHexString } from './helpers'
+import { bytesToHex, bytesToUtf8, hexToBytes } from 'ethereum-cryptography/utils'
+
+import { assertIsArray, assertIsBytes, assertIsHexString } from './helpers'
 import { isHexPrefixed, isHexString, padToEven, stripHexPrefix } from './internal'
 
 import type {
   NestedBufferArray,
   NestedUint8Array,
   PrefixedHexString,
-  TransformableToArray,
-  TransformableToBuffer,
+  TransformabletoBytes,
 } from './types'
 
+/****************  Borrowed from @chainsafe/ssz */
+// Caching this info costs about ~1000 bytes and speeds up toHexString() by x6
+const hexByByte = new Array<string>(256)
+
+export const bytesToPrefixedHexString = (bytes: Uint8Array): string => {
+  let hex = '0x'
+  for (const byte of bytes) {
+    if (!hexByByte[byte]) {
+      hexByByte[byte] = byte < 16 ? '0' + byte.toString(16) : byte.toString(16)
+    }
+    hex += hexByByte[byte]
+  }
+  return hex
+}
+
+export const hexStringToBytes = (hex: string): Uint8Array => {
+  if (typeof hex !== 'string') {
+    throw new Error(`hex argument type ${typeof hex} must be of type string`)
+  }
+
+  if (hex.startsWith('0x')) {
+    hex = hex.slice(2)
+  }
+
+  if (hex.length % 2 !== 0) {
+    throw new Error(`hex string length ${hex.length} must be multiple of 2`)
+  }
+
+  const byteLen = hex.length / 2
+  const bytes = new Uint8Array(byteLen)
+  for (let i = 0; i < byteLen; i++) {
+    const byte = parseInt(hex.slice(i * 2, (i + 1) * 2), 16)
+    bytes[i] = byte
+  }
```

Comment:
Can we not refactor this part with `hexToBytes`?

Reply:
We absolutely can.  As noted above, I lifted all of this code directly from chainsafe (and tried to bracket it off so we were aware).  I think we should do some benchmarking between the `hexToBytes` method and this one.  I think the biggest thing I was trying to delineate was the difference beteween prefixed hex strings and raw hex.  Whether we call them "prefixed hex strings" where we expect a `0x` and just "hex" where we don't, we should be consistent on this point.  Then, as long as we consistently use the same functions for prefixed hex strings versus raw hex, it should be a simple matter to benchmark the implementations and then just swap them out if we find a faster one.